Enable new email templates for existing stores without email customization

NeosinneR opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

1. Enable new email templates (`Email improvements` feature) for existing merchants who don't have email templates overridden and don't use email customizer plugin. 
2. Adds a modal on **WooCommerce > Home** and a note in **WooCommerce Inbox**  when the feature is enabled, and another one when the feature is not enabled to try it out. 
3. Allows trying out new email templates in the **WooCommerce > Settings > Emails** without the need to actually enable the feature. 
4. Show new email settings for everyone, but disable fields that require new email templates.

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes WOOPLUG-4203, closes #57819.

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| New feature | Screenshot |
| ------ | ----- |
|   Inbox note     |    <img width="759" alt="Screenshot 2025-05-11 at 23 36 42" src="https://github.com/user-attachments/assets/ea35578f-c9a2-4c71-ba6b-430cdae6e07c" />   |
|   Home modal     |     <img width="1113" alt="Screenshot 2025-05-11 at 23 37 03" src="https://github.com/user-attachments/assets/2f71bf23-08d7-48f3-bf13-d49d71a0ebb3" />  |
|   Trying out new templates    |   <img width="676" alt="Screenshot 2025-05-11 at 23 37 25" src="https://github.com/user-attachments/assets/dd66ed74-ff4f-4e4a-a5ca-091ebe21ef81" />    |
|   Disabled email settings fields     |   <img width="1103" alt="Screenshot 2025-05-11 at 23 38 02" src="https://github.com/user-attachments/assets/81c763d4-361e-40e0-bb90-38055fab8a79" />    |

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set up 5 different websites on Jurassic.ninja
    1. One with WooCommerce 9.8 - should have `Email improvements` feature auto-enabled.
    2. One with WooCommerce 9.7 - should have `Email improvements` feature disabled.
    3. One with WooCommerce 9.8 - But manually disable `Email improvements` in **WooCommerce > Settings > Advanced > Features**.
    5. One with WooCommerce 9.7 and any email template overridden (go to **WooCommerce > Settings > Emails** and manage any email. On this page, click `Copy file to theme`).
    6. One with WooCommerce 9.7 and any email customizer plugin installed and active.
2. Update each site with the build from this PR.
    1. There should be no change, no modal on **WooCommerce > Home**, and no note in WooCommerce Inbox.
    2. The feature should be auto-enabled. On **WooCommerce > Home**, there should be a modal with a `Manage emails` link. 
    3. The feature should be disabled. On **WooCommerce > Home**, there should be a modal with a `Try new templates` link. Clicking it should show a preview of new emails. Similarly, a note in the Inbox.
    4. The same as website 3.
    5. The same as website 3.
3. The notice on homepage in inbox section is displayed after the `wc_admin_daily` cron run. You can install WP Crontrol plugin to trigger this cron.
4. On all websites except 1., at the bottom of **WooCommerce > Settings > Emails**, you should see a button to either try new templates or revert to the legacy one, depending on the status of the feature.
5. After reverting to legacy emails, you should see a feedback modal. 
6. When visiting the **WooCommerce > Home**, you should at most see the modal once (though you can always force it by adding `emailImprovementsModal=try` or `emailImprovementsModal=enabled` into URL).

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

NeosinneR committed:
Always show the same email template description

NeosinneR committed:
Always show logo selector in email template settings

NeosinneR committed:
Always use new naming for email template colors

NeosinneR committed:
Always show Color palette section in Emails settings

NeosinneR committed:
Update default footer text and available placeholders

NeosinneR committed:
Show new email settings as disabled when feature is not enabled

NeosinneR committed:
Show email improvements in Settings when URL parameter is present

NeosinneR committed:
Pass temporary email improvements to email preview

NeosinneR committed:
Don't allow "Sync with theme" when email improvements are disabled

This is because the old email templates are not updated for new color schemes

NeosinneR committed:
Add missing doc comment

NeosinneR committed:
Fix return type

NeosinneR committed:
More descriptive method name

NeosinneR committed:
Add EmailOverrides class with helper methods

NeosinneR committed:
Properly enable email_improvements for new installations

NeosinneR committed:
Enable email improvements by default for existing shops if conditions are met

NeosinneR committed:
Remove no longer needed spacing for email preview

NeosinneR committed:
Add inbox note for email improvements

NeosinneR committed:
Create method to check if trying new templates

NeosinneR committed:
Add info notice when previewing new templates

NeosinneR committed:
Enable email improvements when settings are saved while trying new templates

NeosinneR committed:
Add buttons to try email templates and revert back to legacy

NeosinneR committed:
Add confirmation modal before reverting to legacy email templates

NeosinneR committed:
Show feedback form when reverting to legacy templates

This is reusing the same form when the feature is disabled in Settings > Afvanced > Features

NeosinneR committed:
Better naming for email improvements class

NeosinneR committed:
Move decision making about email improvements into a single class

NeosinneR committed:
Add email improvements modal to homepage

NeosinneR committed:
Show email improvements modal on WC home page

NeosinneR committed:
Dismiss try email improvements modal when already trying out in Settings

NeosinneR committed:
Add changelog

NeosinneR committed:
Update tests

costasovo commented on plugins/woocommerce/src/Internal/Admin/EmailPreview/EmailPreview.php:
```diff
@@ -30,6 +30,8 @@ class EmailPreview {
 		'WC_Email_Customer_Reset_Password',
 	);
 
+	const TRANSIENT_PREVIEW_EMAIL_IMPROVEMENTS = 'preview_email_improvements';
```
I'm a bit nitpicking here, but shouldn't we prefix the transient with `woocommerce_`? Woo seems to use such prefix in other cases ([example](https://github.com/woocommerce/woocommerce/blob/c46ae897aa0e2bb0f3290416967f60d7246b1e23/plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php#L15)).

costasovo commented on plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php:
```diff
@@ -865,6 +866,20 @@ public function track_email_improvements_feature_change( $feature_id, $enabled )
+			$feature_controller = wc_get_container()->get( FeaturesController::class );
+			$feature_controller->change_feature_enable( 'email_improvements', true );
+
+			// Remove the try-new-templates parameter from the URL.
+			wp_safe_redirect( remove_query_arg( 'try-new-templates' ) );
+			exit;
```
I have a concern here that if multiple plugins hook to `woocommerce_settings_saved` calling `exit;` will prevent executing their registered actions.
Do we need to remove the parameter? Or perhaps we could register the action with a very low priority to run as the last.

NeosinneR replied:
Good catch! It's somewhat important to remove the parameter, because otherwise you could click on save settings (which enables the feature) and then click on `Revert to legacy templates` but because the parameter would still be in the URL, visually you'd still see new templates and not the reverted one - which would look like a bug.

Technically, I could remove the parameter in a different part of the code, but that would add complexity as to why it's called there and not immediately after it's no longer needed.

costasovo commented on plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php:
```diff
@@ -828,6 +835,40 @@ public function previewing_new_templates(): void {
+			);
+			?>
+			<hr class="wc-settings-email-color-palette-separator" />
+			<a href="<?php echo esc_url( add_query_arg( $disable_feature_args ) ); ?>" class="components-button is-tertiary">
```
![Screenshot 2025-05-12 at 10 46 38](https://github.com/user-attachments/assets/5218375d-4dd2-4371-b127-09677f51980c)
The "Revert to legacy" button looks misaligned. We could make it render like a link `is-link` or make it a secondary button, which has border.

costasovo commented on plugins/woocommerce/src/Internal/Features/FeaturesController.php:
```diff
@@ -1646,6 +1646,7 @@ public function change_feature_enable_from_query_params(): void {
 	 */
 	public function display_email_improvements_feedback_notice( $feature_id, $is_enabled ): void {
 		if ( 'email_improvements' === $feature_id && ! $is_enabled ) {
+			set_transient( 'wc_settings_email_improvements_reverted', 'yes', 15 );
```
The 15 seconds explain why I got the form twice when I cancelled it and reloaded the page. But I was testing and clicking fast so I assume that it is fine. If we want to prevent this we could delete the transient once we render the feedback slotfill, but I consider it an optional improvement.

NeosinneR replied:
I don't know why, but when I try to `delete_transient()` in the condition where the feedback slotfill is rendered, it never renders. I can't find the reason why, but considering this shouldn't be the usual case, and even if you see the feedback modal twice, there is no business harm, I kept it like this.

costasovo reviewed (commented):
@NeosinneR That's a huge PR! Good job! I added a couple of comments with some ideas and questions, but nothing super critical.

I'll add a report from testing in another comment, but it all worked as expected.

NeosinneR committed:
Prefix transient value name

NeosinneR committed:
Use link instead of tertiary button to try/revert new email templates

NeosinneR committed:
Remove "try-new-templates" parameter after enabling the feature as a low priority

This is to increase the chance, that if other extensions hook into into "woocommerce_settings_saved", they will be executed before "exit;" is called

NeosinneR commented:
Thanks for the comprehensive review, @costasovo, I really appreciate it on such short notice. I've addressed three of four of your suggestions while keeping the last one unchanged (explained in the comment above).

costasovo reviewed (approved):
@NeosinneR Thank you for the updates!

I retested scenario 3 and all still works as expected.
The button is now aligned.
![Screenshot 2025-05-12 at 15 02 12](https://github.com/user-attachments/assets/7c79523b-3a6c-463a-9c42-c0349069aea7)

I see the performance job failed. I guess it needs a re-run. Once green we can merge.
